pyo3-macros-backend/src/stubs.rs

```diff
@@ -0,0 +1,67 @@
+use syn::{GenericArgument, PathArguments, Type};
+
+pub fn map_rust_type_to_python(rust: &Type) -> String {
```

Comment:
Rather than embedding this information into the macro, it would be cool if we could somehow use a trait to calculate this. I think that'd mean code generating a function (maybe as part of `#[pymodule]`) which could be run to build subs?

Reply:
Indeed, but I have no idea where to extract that information from.
I liked the compile-time approach because it would be easier to transparently include into Maturin (it could automatically enable the cargo feature, collect the generated files, merge them together and continue on without any code modifications), but I'm not yet sure that all information is available then.

Is your idea something like this?
```rust
trait PythonType {
    fn python_type() -> &'static str
}
```
It would probably be easy to implement for all built-in PyO3 types, and I guess `#[pyclass]` and `#[derive(FromPyObject)]` can generate it as well.

I'm not sure how the other needed information (full method signature, etc) can be passed to the runtime though.